schema/crdb/crucible-ref-count-records/up08.sql

```diff
@@ -0,0 +1,31 @@
+/* Construct a UUIDv4 deterministically from the existing data in the region
+   snapshot records to populate the volume resource usage records */
+INSERT INTO volume_resource_usage (
+  SELECT
+    (
+      OVERLAY(
+       OVERLAY(
+        MD5(volume.id::TEXT || dataset_id::TEXT || region_id::TEXT || snapshot_id::TEXT || snapshot_addr || volume_references::TEXT)
+        PLACING '4' from 13
+       )
+       PLACING '8' from 17
```

Comment:
This still feels somewhat dangerous to me, and I'm not sure we have tests for the validity nor uniqueness of the UUIDs being produced by this transition.

The goal here is to produce a `volume_resource_usage` record for each `volume INNER JOIN region_snapshot` that isn't being deleted, right? Couldn't you do this by doing:

```sql
-- End goal: Find all volume/region_snapshot combos that **don't** have a volume_resource_usage record
(volume INNER JOIN region_snapshot) LEFT JOIN (volume_resource_usage)
  ON
(
  volume_resource_usage.volume = volume.id AND
  volume_resource_usage.region_snapshot_region_id = region_snapshot.region_id
)
```

And then write a `WHERE` clause to ensure that  `volume_resource_usage.id` is NULL for these cases?

Basically:

```sql
INSERT INTO  volume_resource_usage
-- <New records, using normal UUID generation>
FROM
-- <The clause above - `volume JOIN region_snapshot`, but also `JOIN`-ing with `volume_resource_usage` to check a record doesn't already exist>
WHERE
volume.time_deleted IS NULL AND
region_snapshot.deleting = false AND
volume_resource_usage.id IS NULL
```

This would still be idempotent! Re-running it should be a no-op on the second iteration

Reply:
This does work, see 2b511b17d, thanks :) The `test_migrate_.*` tests confirm that it works.